src/api/internal/users/repository/repository.go

```diff
@@ -124,18 +125,37 @@ func (r *UserRepositoryImpl) ListUsers(ctx context.Context, search string, sortD
 	return users, nil
 }
 
-func (r *UserRepositoryImpl) GetUserByEmailAndPassword(ctx context.Context, email, password string) (*domain.User, error) {
-	query := `
-		SELECT uuid, name, email, picture, phone, role
-		FROM users
-		WHERE email = @email AND password = @password
-	`
-	row := r.DB.QueryRowContext(ctx, query, sql.Named("email", email), sql.Named("password", password))
+func (r *UserRepositoryImpl) GetUserByEmailAndPassword(ctx context.Context, email, password *string, strictMode bool) (*domain.User, error) {
```

Comment:
I would split this method into two

GetUserByEmail(ctx, email)

AuthenticateUser(ctx, email, password) err